### ISSUE TITLE
library_bootstrap is not resumable; mid-walk crash loses all progress

### ISSUE THREAD

**Issue by networkingguru** [networkingguru/music-discovery#31]
## Symptom

`artist_id_migration.library_bootstrap` walks 62k library tracks, makes ~10-20k MusicKit API calls (one search + one song-fetch per unique (artist, track) pair), and writes `track_persistent_id_cache.json` + `bootstrap_complete.sentinel` ONLY at the end of a successful full walk:

```python
# artist_id_migration.py:155-181
tracks = collect_library_tracks_jxa()
cache: dict[str, ArtistRef] = {}
seen: dict[...] = {}
for t in tracks:
    ...
    cache[pid] = ref
cache_dir.mkdir(parents=True, exist_ok=True)
save_track_persistent_id_cache(cache, cache_dir / "track_persistent_id_cache.json")
(cache_dir / "bootstrap_complete.sentinel").write_text("1")
return n_calls
```

If anything raises mid-walk — network reset (#30), JXA timeout, transient API issue — the entire 1.5-3h of work is discarded. The cache file isn't written, the sentinel isn't written, and re-running starts from zero.

## Reproduction

Dry-run #3 + dry-run #4 both crashed during bootstrap with `ConnectionResetError` (#30). Each loss = 1.5-3h of API work + wall time, regardless of how close to completion they were.

## Fix shape

Add periodic checkpointing inside the bootstrap loop:

1. At start of `library_bootstrap`, load any existing `track_persistent_id_cache.json` as `cache` (if file exists from a prior partial run). Skip pids that are already cached.
2. Pre-populate the in-memory `seen[(artist, name)]` dedup map from `cache` by walking JXA tracks once and matching pids → existing refs. Avoids re-API-calling (artist, track) pairs whose persistentIDs we already cached.
3. Every N successful resolutions (e.g. N=500), atomically rewrite `track_persistent_id_cache.json`. So a crash loses at most N track resolutions, not the whole bootstrap.
4. Final save + sentinel only on complete walk (no change).

Pairs naturally with #30: even with network-retry in place, hard failures still happen (kernel TCP queue overflow, signal interrupts, JXA timeout, etc.). Resumability is the safety net.

Tests:
- New test: pre-populate `track_persistent_id_cache.json` with N entries, run bootstrap, verify it resumes (skips already-cached pids).
- New test: force a crash mid-walk via patched resolve_library_track, restart, verify final state matches a clean run.
- Existing tests should keep passing (single-run case).

Surfaced during Phase 3 dry-run validation 2026-05-04/05.

**Comment by networkingguru** [comment]
Fixed in c0b3c3c: library_bootstrap checkpoints and resumes (artist_id_migration.py ~145-186). Migration-only path. Audit 2026-09-07.
